httputil/client.go

```diff
@@ -33,6 +41,58 @@ type ClientOptions struct {
 	Proxy      func(*http.Request) (*url.URL, error)
 }
 
+func addLocalSSLCertificates(conf *tls.Config) (allCAs *x509.CertPool, err error) {
+	if conf != nil && conf.RootCAs != nil {
+		allCAs = conf.RootCAs
+	} else {
+		allCAs, err = x509.SystemCertPool()
+		if err != nil {
+			return nil, fmt.Errorf("cannot read system certificates: %v", err)
+		}
+	}
+	if allCAs == nil {
+		return nil, fmt.Errorf("cannot use empty certificate pool")
+	}
+	extraCertFiles, err := filepath.Glob(filepath.Join(dirs.SnapdExtraSSLCertsDir, "*.pem"))
+	if err != nil {
+		return nil, err
+	}
+	for _, p := range extraCertFiles {
+		// XXX: cache these reads?
+		extraCert, err := ioutil.ReadFile(p)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read extra certificate: %v", err)
+		}
+		if ok := allCAs.AppendCertsFromPEM(extraCert); !ok {
+			logger.Noticef("cannot append store ssl certificate: %v", p)
+		}
+	}
+	return allCAs, nil
+}
+
+// dialTLS holds a tls.Config that is used by the dialTLS.dialTLS()
+// function.
+type dialTLS struct {
+	conf *tls.Config
+}
+
+// dialTLS will use it's tls.Config and use that to do a tls connection.
+func (d *dialTLS) dialTLS(network, addr string) (net.Conn, error) {
+	if d.conf == nil {
+		// c.f. go source: crypto/tls/common.go
+		var emptyConfig tls.Config
+		d.conf = &emptyConfig
```

Comment:
Furthermore, Ubuntu is also in the process of disabling < 1.2. https://launchpad.net/bugs/1856428

I didn't mention it before, but there is also tls.MaxVersion. You could use this to setup a test server and verify the client can't connect (where the server is otherwise configured with valid certs).

Reply:
Sure, I added code for this now.